pkg/aggregator/aggregator.go

```diff
@@ -354,11 +358,46 @@ func NewBufferedAggregator(s serializer.MetricSerializer, eventPlatformForwarder
 		flushAndSerializeInParallel: NewFlushAndSerializeInParallel(pkgconfigsetup.Datadog()),
 
 		filterListChan: make(chan utilstrings.Matcher),
+		tagFilterList:  loadTagFilterList(),
 	}
 
 	return aggregator
 }
 
+func loadTagFilterList() *filterlist.TagMatcher {
+	tagFilterListInterface := pkgconfigsetup.Datadog().GetStringMap("metric_tag_filterlist")
+
+	tagFilterList := make(map[string]filterlist.MetricTagList, len(tagFilterListInterface))
+	for metricName, tags := range tagFilterListInterface {
+		// Tags can be configured as a simple array of tags - default to exclude.
+		arr, ok := tags.([]string)
+		if ok {
+			tagFilterList[metricName] = filterlist.MetricTagList{
+				Tags:   arr,
+				Action: "exclude",
+			}
+		} else {
+			// Or tags can be configured as an object with fields:
+			// tags - array of tags
+			// action - either `include` or `exclude`.
+			// Roundtrip the struct through yaml to load it.
+			tagBytes, err := yaml.Marshal(tags)
```

Comment:
Other parts of the code use `github.com/go-viper/mapstructure/v2` for the same function, would it make sense to use it here as well?